tests/nominal/test_OneHotEncodingTransformer.py

```diff
@@ -106,7 +106,6 @@ def create_OneHotEncoderTransformer_test_df_2():
                 "b": ["w", "w", "z", "y", "x"],
                 "c": ["a", "a", "c", "b", "a"],
             },
-            index=[10, 15, 200, 251, 59],
```

Comment:
so this was just quite a strict test that abnormal index columns are preserved. But polars doesn't really fuss with default index columns in the same way pandas does so can't really test in the same way.
https://docs.pola.rs/user-guide/migration/pandas/

So that we're not losing any test coverage, I've now added a generic pandas only test that looks at a weird index df
https://github.com/lvgig/tubular/pull/340/commits/94a5d00e415e1a19544be83cd889b45917a95382